libs/ledger-live-common/src/currencies/support.ts

```diff
@@ -97,5 +97,5 @@ export function listSupportedCurrencies(): CryptoCurrency[] {
     : userSupportedCurrencies.concat(experimentals);
 }
 export function isCurrencySupported(currency: CryptoCurrency): boolean {
-  return listSupportedCurrencies().includes(currency);
+  return listSupportedCurrencies().some((c) => c.id === currency.id);
```

Comment:
what problem required to change this? normally a currency don't exist twice with different reference so this should never be needed.

Reply:
So something is weird when I fetch so reference doesn't seems to match, I'll try to dig more about this then

Reply:
I am getting a lot of confusion about this one and i require some assistances.

I am fetch the tokens from preload and use `convertERC20` from  `"@ledgerhq/cryptoassets` : 

`libs/ledger-live-common/src/families/ethereum/modules/erc20.ts` : 
```
tokens = data.map(convertERC20);
```

and convertERC20 token apply this parentCurrency : 
```
  const parentCurrency = getCryptoCurrencyById(parentCurrencyId);
 ```

and then once fetch I'm doing a simple  `addTokens(tokens)` from the same methods as other tokens.

But I still don't understand why the reference between those two are different. Is it possible because I did it by exporting it will change ?

```
      console.log(getCryptoCurrencyById("ethereum"));
      console.log(t.parentCurrency);
      console.log("same ? ", getCryptoCurrencyById("ethereum") === t.parentCurrency); // false
```

Reply:
We tried to investigate it but It's seems that the references between these two cryptocurrencies are different because they may come from different thread.

We think when we do a preload that add token in the renderer thread it somehow create an another refererence of ethereum which can be compare with the one from isCurrencySupported.


I'm not sure if we can conciliate those two things, if you think we can I'll be glad to have some pair pro about this.

We come with a quick fix like this : 
```
  return (
    listSupportedCurrencies().includes(currency) ||
    listSupportedCurrencies().some((c) => c.id === currency.id)
  );
  ```

but the same problem remains.

Reply:
we absolutely need to solve this, these workaround shouldn't be made, because we will have 100 other places where we expect this to work correctly. we need to fix the root cause.

Reply:
possible causes:
- module imported twice
  - duplicate package?
  - lib vs lib-es ?
- currency object is getting recreated somewhere. `{...currency}` or something?

to me it have nothing to do with threads, we never send a "currency" object through thread neither

Reply:
spotted issue: `preload` must not be returned the "tokens" straightaway, these aren't to be "serialized" because they contain parentCurrency plain objects, you'll need to introduce a "raw token currency" type to make the intermediary object that can be "cached"

Reply:
Thank you ! It was indeed a lot better solution this way !